lug-helper.sh

```diff
@@ -2083,14 +2118,19 @@ rm_shaders() {
     fi
 
     # Loop through all possible shader directories
-    for appdata_dir in "$shaders_dir"/*; do
-        if [ -d "$appdata_dir/$shaders_subdir" ]; then
-            # If a shaders directory is found, delete it
-            if message question "The following directory will be deleted:\n\n$appdata_dir/$shaders_subdir\n\nDo you want to proceed?"; then
-                debug_print continue "Deleting $appdata_dir/$shaders_subdir..."
-                rm -r "${appdata_dir:?}/$shaders_subdir"
+    for appdata_dir in "$wine_prefix/$appdata_path"/$appdata_dir_mask; do
```

Comment:
Giving everything a final look-over and I was wondering about this directory name mask. Is it necessary? Without it, we check every directory in the Star Citizen appdata dir for a shaders subdir. There's not too much going on in the appdata directory so this doesn't seem too expensive to me, and would be more future proof for if/when they change how they store shaders again.

So, real question: What problem am I missing that this was solving?

Reply:
I believe the idea may have been to evade a long loop of delete confirmation dialogs if there ever were more types of directories in there but that may have been made before understanding the dialog for each one. I'm open to letting it loop everything

Reply:
> evade a long loop of delete confirmation dialogs if there ever were more types of directories in there

If we have to clear out other types of directories, I imagine we'd be handling that separately from this delete shaders operation. So this one would only ever be prompting to delete the shaders directories anyway.

>  the other idea was to preserve the vulkan setting by not deleting the entire parent sc-alpha-* directory

So in case you missed it, only the shaders subdirectories are deleted. The parent sc-alpha directory remains behind afterwards, because we can't be certain that they won't add other directories besides shaders to it in the future. My install right now has a new GraphicsSettings directory, so it appears they've decided to start doing just that. I generally try my best to future-proof this kind of thing so we don't have to worry about that situation.

> for that matter a menu item to switch between dx11 and vulkan might be preferable to this whitelist of directory names

It prompts the user for each directory deletion, so it's already possible for the user to selectively delete specific old game version shaders or just the dx11 ones. This does assume the user knows that "Shaders" means dx11 though. The vulkan directory is at least obvious.